templates/multisig/test/multisig.js

```diff
@@ -26,283 +27,363 @@ const ExecutionTarget = artifacts.require('ExecutionTarget')
+            })
+          })
+        }
       })
 
-      context('> Voting access', () => {
-        it('set up voting correctly', async () => {
-          assert.equal((await voting.voteTime()).toString(), VOTING_TIME, 'voting time not correct')
-          assert.equal((await voting.supportRequiredPct()).toString(), REQUIRED_SUPPORT.toString(), 'support required not correct')
-          assert.equal((await voting.minAcceptQuorumPct()).toString(), REQUIRED_SUPPORT.toString(), 'accept quorum not correct')
+      context('when the creation succeeds', () => {
+        before('create multisig entity', async () => {
+          if (creationStyle === 'single') {
+            instanceReceipt = await template.newTokenAndInstance(TOKEN_NAME, TOKEN_SYMBOL, daoID, SIGNERS, REQUIRED_SIGNATURES)
+            tokenReceipt = instanceReceipt
+          } else if (creationStyle === 'separate') {
+            tokenReceipt = await template.newToken(TOKEN_NAME, TOKEN_SYMBOL)
+            instanceReceipt = await template.newInstance(daoID, SIGNERS, REQUIRED_SIGNATURES)
+          }
+
+          dao = Kernel.at(getEventArgument(instanceReceipt, 'DeployDao', 'dao'))
+          token = MiniMeToken.at(getEventArgument(tokenReceipt, 'DeployToken', 'token'))
         })
 
-        it('cannot reinitialize voting', async () => {
-          await assertRevert(voting.initialize.request(token.address, 1e18, 1e18, 1000), 'INIT_ALREADY_INITIALIZED')
+        before('load apps', async () => {
+          const installedApps = getInstalledAppsById(instanceReceipt)
+          assert.equal(installedApps.vault.length, 1, 'should have installed 1 vault app')
+          assert.equal(installedApps.voting.length, 1, 'should have installed 1 voting app')
+          assert.equal(installedApps.finance.length, 1, 'should have installed 1 finance app')
+          assert.equal(installedApps['token-manager'].length, 1, 'should have installed 1 token manager app')
+
+          acl = ACL.at(await dao.acl())
+          vault = Vault.at(installedApps.vault[0])
+          voting = Voting.at(installedApps.voting[0])
+          finance = Finance.at(installedApps.finance[0])
+          tokenManager = TokenManager.at(installedApps['token-manager'][0])
         })
 
-        it('fails trying to modify support threshold directly', async () => {
-          await assertRevert(voting.changeSupportRequiredPct.request(REQUIRED_SUPPORT.add(1), { from: owner }), 'APP_AUTH_FAILED')
+        it('costs ~6.6e6 gas', async () => {
+          if (creationStyle === 'single') {
+            assert.isAtMost(instanceReceipt.receipt.gasUsed, 6.7e6, 'create script should cost almost 6.7e6 gas')
+          } else if (creationStyle === 'separate') {
+            assert.isAtMost(tokenReceipt.receipt.gasUsed, 1.8e6, 'create token script should cost almost 1.8e6 gas')
+            assert.isAtMost(instanceReceipt.receipt.gasUsed, 5e6, 'create instance script should cost almost 5e6 gas')
```

Comment:
Interesting, this is a good sign, at least. The old kit used to require 5.7M gas to do just the instance (no token), so it looks like we've shaved off a bit with the new base template 👍